File: app/src/main/java/com/johnnyzhou/movieme/ui/common/BaseActivity.java
package com.johnnyzhou.movieme.ui.common;

import androidx.appcompat.app.AppCompatActivity;

import com.johnnyzhou.movieme.MovieMeApp;
import com.johnnyzhou.movieme.di.component.AppComponent;

public abstract class BaseActivity extends AppCompatActivity {
    public AppComponent getApplicationComponent() {
        return ((MovieMeApp) getApplication()).getAppComponent();
    }
}

File: app/src/main/java/com/johnnyzhou/movieme/ui/common/BaseActivity.kt
package com.johnnyzhou.movieme.ui.common

import androidx.appcompat.app.AppCompatActivity
import com.johnnyzhou.movieme.MovieMeApplication
import com.johnnyzhou.movieme.di.component.AppComponent

abstract class BaseActivity : AppCompatActivity() {
    val applicationComponent: AppComponent
        get() = (application as MovieMeApplication).appComponent
}